src/api.ts

```diff
@@ -84,26 +83,25 @@ router.post(
         return res.status(400).send(validateBidPayloadSchema.errors);
       }
       const dto: BidPayloadPostDto = req.body as BidPayloadPostDto;
-      // Generate bidNonce, nftId
-      const nftId = calculateNftId(dto.contractAddress, dto.tokenId);
+
+      // Generate bidNonce
       const bidNonce = Math.floor(Math.random() * 42949672960);
 
       // We use `bidNonce` to be clearer about what the variable actually represents
-      // but any existing database records will still show `bidNonce`
+      // but any older database records may still show `auctionId`
       const payload = await encodeBid(
         bidNonce,
         dto.bidAmount,
-        dto.contractAddress,
         dto.tokenId,
         dto.minimumBid,
         dto.startBlock,
-        dto.expireBlock
+        dto.expireBlock,
+        dto.bidToken
```

Comment:
We should call the contract and verify that `dto.bidToken` is a valid token to be used in this scenario

Reply:
Good idea